core/api/service/api_service.cpp

```diff
@@ -71,4 +161,21 @@ namespace kagome::api {
     logger_->debug("Service stopped");
   }
 
+  outcome::result<uint32_t> ApiService::subscribe_thread_session_to_keys(
+      const std::vector<common::Buffer> &keys) {
+    if (auto session_id = threaded_info.fetch_thread_session_id(); session_id) {
+      if (auto session = find_session_by_id(*session_id)) {
+        for (auto &key : keys) {
+          /// TODO(iceseer): make move data to subscription
```

Comment:
Task id, please.

Reply:
No task id. It's a part of optimization process.